virttest/qemu_devices/qcontainer.py

```diff
@@ -3094,31 +3094,31 @@ def tpm_define_by_params(self, name, params):
         tpm_type = params.get("tpm_type")
         tpm_version = params.get("tpm_version")
 
-        tmp_dir = data_dir.get_tmp_dir()
-        random_str = utils_misc.generate_random_string(6)
+        swtpm_dir = os.path.join(data_dir.get_data_dir(), "swtpm")
 
         def _handle_log(line):
             try:
-                utils_misc.log_line('%s_%s_swtpm_setup_%s.log' %
-                                    (self.vmname, name, random_str), line)
+                utils_misc.log_line('%s_swtpm_setup.log' % name, line)
             except Exception as e:
-                LOG.warn("Can't log %s_%s_swtpm_setup output: %s.", self.vmname, name, e)
+                LOG.warn("Can't log %s_swtpm_setup output: %s.", name, e)
 
         def _emulator_setup(binary, extra_options=None):
             setup_cmd = binary
             if tpm_version in ('2.0',):
                 setup_cmd += ' --tpm2'
 
-            tpm_path = os.path.join(tmp_dir, '%s_%s_tpm_state_%s' %
-                                    (self.vmname, name, random_str))
-            if os.path.exists(tpm_path):
-                shutil.rmtree(tpm_path, ignore_errors=True)
-            os.makedirs(tpm_path)
+            tpm_path = os.path.join(swtpm_dir, '%s_state' % name)
```

Comment:
For migration case, maybe we need to make the path to be unique.

Reply:
Yes, a good catch. But as you know, previous vTPM migration is unique but it is not the same vTPM, it can work but is not a good idea. So I will keep this design, and I have already pushed a new commit. To workaround migration after using the namespace to manage all vTPM devices.

Reply:
Migration works for me now. 

As we discussed, need patch for multi guest test case to make it work.

Reply:
Run some Win11 migration tests on single host:

 (1/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_live_migration.before_bg_test.q35: STARTED
 (1/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_live_migration.before_bg_test.q35: PASS (192.41 s)
 (2/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_live_migration.during_bg_test.q35: STARTED
 (2/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_live_migration.during_bg_test.q35: PASS (200.06 s)
 (3/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_live_migration.after_bg_test.q35: STARTED
 (3/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_live_migration.after_bg_test.q35: PASS (190.79 s)
 (4/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_live_migration.before_bg_test.q35: STARTED| (4/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_live_migration.before_bg_test.q35: PASS (159.20 s)
 (5/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_live_migration.during_bg_test.q35: STARTED
 (5/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_live_migration.during_bg_test.q35: PASS (175.09 s)
 (6/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_live_migration.after_bg_test.q35: STARTED
 (6/6) Host_RHEL.m9.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_live_migration.after_bg_test.q35: PASS (188.52 s)